### PR TITLE
Normalize changeset spec fields to speed up joins

### PR DESCRIPTION
**Pull request by mrnugget**
This is the result of @eseliger and me running into really slow `GetRewirerMapping` queries yesteday.

We found out that a single `changeset_spec` with an abnormally large `spec` column was the cause of this performance regression:

1. The `spec->'diff'` contained ~57mil lines of code.
2. The previous `GetRewirerMapping` query caused the database to apparently load the complete JSONB columns from disk.

57 million lines in a single diff is not normal, but it was the result of accidentally adding the `node_modules` folder to the diff. And *that* is not that uncommon.

And such a single, accidentally-uploaded changeset spec caused performance problems for the complete instance.

So we decided to normalise the 3 `spec` fields that we use for JOINs and WHERE conditions by putting them in separate columns, with indexes.

No performance problems anymore and the queries got quite a bit cleaner too.

**Note** though that these columns are **write-only** for now and are only read by SQL queries. We didn't see a need to make use of them in application code for now, but that may change in the future at which point we should add them to the `ChangesetSpec` type definition.

### CODE REVIEW

**Review by eseliger — APPROVED** [review]
Excellent work 🌟 

But we should add an explanation why this change is good. You or me?

**Comment by mrnugget** [comment]
> But we should add an explanation why this change is good. You or me?

Feel free to push to this PR! Otherwise I'd try to finish it tomorrow. Also haven't requested a review yet :)

File: migrations/frontend/1528395782_normalize_spec_fields_on_changeset_specs.up.sql [hunk 1]
```diff
@@ -0,0 +1,80 @@
+BEGIN;
+
+ALTER TABLE changeset_specs ADD COLUMN head_ref text DEFAULT NULL;
+ALTER TABLE changeset_specs ADD COLUMN title text DEFAULT NULL;
+ALTER TABLE changeset_specs ADD COLUMN external_id text DEFAULT NULL;
+
+CREATE INDEX changeset_specs_external_id ON changeset_specs (external_id);
+CREATE INDEX changeset_specs_head_ref ON changeset_specs (head_ref);
+CREATE INDEX changeset_specs_title ON changeset_specs (external_id);
```
Comment (eseliger) [inline migrations/frontend/1528395782_normalize_spec_fields_on_changeset_specs.up.sql:9]:
```suggestion
CREATE INDEX changeset_specs_title ON changeset_specs (title);
```

OH 😨

**Review by eseliger — APPROVED** [review]

File: migrations/frontend/1528395782_normalize_spec_fields_on_changeset_specs.up.sql (hunk 1 above)
Comment (mrnugget) [reply migrations/frontend/1528395782_normalize_spec_fields_on_changeset_specs.up.sql:9]: OHHHH!

Comment (mrnugget) [reply migrations/frontend/1528395782_normalize_spec_fields_on_changeset_specs.up.sql]: Fixed!

**Pull request merged**